### DISCUSSION TITLE
Make spawner mobs count towards PlayerMobCaps

### DISCUSSION THREAD

**Discussion by Thedysons** [PaperMC/Paper#10564] [Ideas]
### Is your feature request related to a problem?

I've noticed that its not uncommon for mobs to build up in dark areas around players AFKing at spawner farms, while the actual mobs spawn rates from the spawners remain unaffected because it seems that mobs from spawners aren't counted towards playermobcaps. Following this logic, this probably also means that spawner mobs can build up indefinitely (or at least until entity cramming takes effect) and from my own testing this seems accurate, while mobs didn't spawn when the spawner was at their level, moving the spawner ~12 above the floor caused them to spawn as normal and build up on the floor. I reached just over 200 mobs before I stopped testing. Provided this behaviour is consistent with other builds/settings, this could be abused to fill the global mob cap (with "count-all-mobs-for-spawning" set to true) or create odd lag machines which are unlikely to be countered by traditional counters.

Going back to the original point, because mobs aren't counted towards playermobcaps, this means players aren't punished for failing to spawn proof their surroundings, and the only option to count them toward globalmobcaps would likely impact players who take the time to spawn proof their area properly. 

### Describe the solution you'd like.

I'm aware than an option "count-all-mobs-for-spawning" exists but from my understanding this applies itself to the global mob cap. This would likely mean that players who take the time to spawn proof their surroundings would see reduced rates so instead I'd like to suggest another option that instead counts mobs from spawners to PlayerMobCaps. I'm not sure if this is technically possible given that the only way mobs from spawners could be attached to playermobcaps is through whose activating the spawner and I'm not sure the API can do that, but I'm sure whoever handles this will know.

This could present another problem where mobs might build up in some tiny cave or dark corner leading to the spawner rates to slowly trickle down so this suggestion could be amended to include an option to despawn the oldest mob in the playersmobcap which could lead to some other issues like pets or museum mobs getting despawned but iirc, named mobs aren't considered in the playermobcounts and would be immune to this measure. 

### Describe alternatives you've considered.

The only option we've considered other than enabling the "count-all-mobs-for-spawning" would be to use a plugin that could disable natural mob spawns but that would quite thoroughly destroy the vanilla experience that we're at least *trying* to emulate.

**Comment by 19MisterX98** [comment]
Why would anyone build spawner farms if they'd still need to spawnproof the area.

**Comment by Thedysons** [comment]
Because they make far more than natural farms? I'm not talking about natural spawners here, the players can get them as items.
